qiskit/chemistry/components/bosonic_basis/harmonic_basis.py

```diff
@@ -0,0 +1,260 @@
+                                    for q in range(p+1):
+                                        coeff = coeff2 * self._harmonic_integrals(
+                                            p, q, indexes[modes[2]], kinetic_term=kinetic_term)
+                                        if abs(coeff) > threshold:
+                                            harmonic_dict[3][modes[0] - 1, m, n,
+                                                             modes[1] - 1, j, k,
+                                                             modes[2] - 1, p, q] += coeff
+                                            if m != n:
+                                                harmonic_dict[3][modes[0] - 1, n, m,
+                                                                 modes[1] - 1, j, k,
+                                                                 modes[2] - 1, p, q] += coeff
+                                            if k != j:
+                                                harmonic_dict[3][modes[0] - 1, m, n,
+                                                                 modes[1] - 1, k, j,
+                                                                 modes[2] - 1, p, q] += coeff
+                                            if p != q:
+                                                harmonic_dict[3][modes[0] - 1, m, n,
+                                                                 modes[1] - 1, j, k,
+                                                                 modes[2] - 1, q, p] += coeff
+                                            if m != n and k != j:
+                                                harmonic_dict[3][modes[0] - 1, n, m,
+                                                                 modes[1] - 1, k, j,
+                                                                 modes[2] - 1, p, q] += coeff
+                                            if m != n and p != q:
+                                                harmonic_dict[3][modes[0] - 1, n, m,
+                                                                 modes[1] - 1, j, k,
+                                                                 modes[2] - 1, q, p] += coeff
+                                            if p != q and k != j:
+                                                harmonic_dict[3][modes[0] - 1, m, n,
+                                                                 modes[1] - 1, k, j,
+                                                                 modes[2] - 1, q, p] += coeff
+                                            if m != n and j != k and p != q:
+                                                harmonic_dict[3][modes[0] - 1, n, m,
+                                                                 modes[1] - 1, k, j,
+                                                                 modes[2] - 1, q, p] += coeff
+            else:
+                raise ValueError('Unexpected order value of {}'.format(order))
```

Comment:
Since order is an internal variable here is there a better way of expressing the message here in terms of the data the user has provided - even saying "Unexpected order of the given hamiltonian" or something though here order to me might be confusing as in order related to sequence where I think order here is related to power order.

Reply:
yes I changed the message to "'Expansion of the PES is too large, only up to 3-body terms are supported"

Reply:
I may misunderstand something here but another error in the `_harmonic_integrals` says `up to 4 is currently supported`.
Are these two different things which are being referred to or is this an inconsistency?

Reply:
yes one is the power of a single mode (4) and the other one is the number of different modes (3), so they are two different aspects linked to the expansion of the potential

Reply:
Okay. Could make that a tiny bit more explicit in the raised errors? After that we can resolve this :slightly_smiling_face:

Reply:
ok

Reply:
I changed the first one "The Q power is to high, only up to 4 is currently supported."

Reply:
theres a reference to a paper there also so it should be pretty explicit